test/livebook/live_markdown/markdown_helpers_test.exs

```diff
@@ -98,6 +148,38 @@ defmodule Livebook.LiveMarkdown.MarkdownHelpersTest do
       assert markdown == reformat(markdown)
     end
 
+    test "alternate headings" do
+      markdown = """
+      Heading 1
+      =========
+
+      Heading 2
+      ---------
+
+      Body\
+      """
+
+      assert markdown == reformat(markdown)
```

Comment:
I'm 90% sure this failure _is_ intentional, since the underline heading format is not terribly useful in Livebook (can only produce `h1` and `h2` which are both special)

These headings are represented with a different `Node` structure inside `earmark-parser`, so this test is valuable to ensure those are handled correctly.

Reply:
Similarly, here reformatting is supposed to normalize to the usual form.

> These headings are represented with a different Node structure inside earmark-parser

Though they are represented as the same AST, and that's the only thing we are dealing with here.